vulnerability-report.md

```diff
@@ -0,0 +1,75 @@
+##Vulnerability Report
+
+###Reviewer: Wenjing Qiang
+
+###Date: May 11, 2016
+
+
+###Injection
+
+Exploiting: Tying singel sql in uploadfile field, programeError will show.
+
+Fix: Pepalce ```curs = connection.cursor() curs.execute( "insert into taskManager_file ('name','path','project_id') values ('%s','%s',%s)" % (name, upload_path, project_id))```
+     with ```taskManager_file = File( name=name, path=upload_path, project=project_id )```
+
+
+###Broken Auth
+
+In user model, there are is_superuser and is_staff fields.  Either to add is_superuser and is_staff in exclude atrribute field or just include what we needed.
+
+Exploiting: If user add checkboxes of "is_superuser" and "is_staff" in singup form in templates, loging as admin will be possible.
+
+Fix : Replace ```exclude = ['groups', 'user_permissions', 'last_login', 'date_joined', 'is_active']```
+         -> ```fields = ['username', 'password', 'email', 'first_name', 'last_name']```
+
+
+###XSS
+
+I found a instance of XSS in forms.py, there is ```|safe``` after username. A javascript prompt can be executed.
+
+Exploiting: Input ```<script>alert("blahblahblah")</script>``` in username.
+
+Fix: Remove |safe
+
+
+###Insecure DOR
+
+In views.py, functions associated with routes doesn't varify user. tasks can be accessed by the url by guessing task id and to edit and delete.
```

Comment:
Show an example. Also, editing and deleting isn't Insecure DOR. Insecure DOR allows this, but the access to the object is all you need.